internal/pkg/api/server/templates/dependencies/dependencies.go

```diff
@@ -184,6 +184,11 @@ func (v *forServer) RepositoryManager() *repository.Manager {
 
 func (v *forServer) EtcdClient() (*etcd.Client, error) {
 	return v.etcdClient.InitAndGet(func() (*etcd.Client, error) {
+		// Check if etcd is enabled
+		if v.Envs().Get("ETCD_ENABLED") == "false" {
+			return nil, fmt.Errorf("etcd integration is disabled")
+		}
+
```

Comment:
Toto sa mi tu stratilo pri rebase, padalo to na hlaske nizsie `ETCD_HOST is not set`.